hspec-core/src/Test/Hspec/Core/Spec/Monad.hs

```diff
@@ -32,16 +32,38 @@ import           Test.Hspec.Core.Tree
 
 import           Test.Hspec.Core.Config.Definition (Config)
 
+-- | A `SpecWith` that can be evaluated directly by the
+-- `Test.Hspec.Core.Runner.hspec` function as it does not require any
+-- parameters.
 type Spec = SpecWith ()
 
+-- | `SpecM` with an empty `ReaderT` environment.
+--
+-- This is the most commonly used spec tree writer type.
+-- It is parameterized over the type @a@, which is the argument type provided
+-- to function-like `Example` instances.
```

Comment:
I think this is not quite right. The `ReaderT` environment is not empty, the `()` is actually the return value of the `SpecM` monad. From a user perspective, the doc is a little too focused on internals and implementation and not use.

I think I'd call it,

```suggestion
-- | A @'SpecWith' a@ represents a test or group of tests that require an @a@ to evaluate.
-- In the common case, a 'Spec' is a @'SpecWith' ()@, which requires @()@ and can be executed with 'hspec'.
-- 
```